src/cache/intern.rs

```diff
@@ -0,0 +1,78 @@
+use cfg_if::cfg_if;
+
+
+cfg_if! {
+    if #[cfg(feature = "enable-interning")] {
+        use std::thread_local;
+        use std::string::String;
+        use std::borrow::ToOwned;
+        use std::cell::RefCell;
+        use std::collections::HashMap;
+        use crate::JsValue;
+        use crate::convert::IntoWasmAbi;
+
+        struct Cache {
+            entries: RefCell<HashMap<String, JsValue>>,
+        }
+
+        thread_local! {
+            static CACHE: Cache = Cache {
+                entries: RefCell::new(HashMap::new()),
+            };
+        }
+
+        /// This returns the raw index of the cached JsValue, so you must take care
+        /// so that you don't use it after it is freed.
+        pub(crate) fn unsafe_get_str(s: &str) -> Option<<JsValue as IntoWasmAbi>::Abi> {
```

Comment:
Note that this is somewhat misleading here and I think this should probably just return `u32`. Below the `into_abi` call isn't called on `JsValue` but is instead called on `&JsValue`. The two trait impls though have the same `Abi` type which is just a `u32`

Reply:
Ah, right, good catch.